relay-server/src/actors/processor.rs

```diff
@@ -2750,22 +2764,38 @@ impl EnvelopeProcessorService {
 impl Service for EnvelopeProcessorService {
     type Interface = EnvelopeProcessor;
 
-    fn spawn_handler(self, mut rx: relay_system::Receiver<Self::Interface>) {
+    fn spawn_handler(mut self, mut rx: relay_system::Receiver<Self::Interface>) {
         let thread_count = self.inner.config.cpu_concurrency();
         relay_log::info!("starting {thread_count} envelope processing workers");
 
         tokio::spawn(async move {
             let semaphore = Arc::new(Semaphore::new(thread_count));
 
-            while let (Some(message), Ok(permit)) =
-                tokio::join!(rx.recv(), semaphore.clone().acquire_owned())
-            {
-                let service = self.clone();
+            let mut global_config_rx = self
+                .inner
+                .global_config
+                .send(Subscribe)
+                .await
+                .expect("EnvelopeProcessorService failed subscribing to GlobalConfigService");
```

Comment:
That might be not a good idea to unwrap here

Reply:
Do you suggest a way to keep going with the select! loop without the global config receiver, or just another way to shut down?

Reply:
Either that, or you even subscribe outside (in ServiceState::create), then clone the receiver, and just pass that in.